biogrid/apps/api/src/app/config/Server.ts

```diff
@@ -0,0 +1,54 @@
+import express from 'express';
+import * as swaggerUi from 'swagger-ui-express';
+import * as bodyParser from 'body-parser';
+import morgan from 'morgan';
+
+import constants from './constants';
+import { ErrorHandler } from './ErrorHandler';
+import { RegisterRoutes } from '../build/routes';
+import { Logger } from './Logger';
+import '../controllers';
+
+export class Server {
+  public app: express.Express = express();
+  private readonly port: number = constants.port;
+
+  constructor() {
+    this.app.use(this.allowCors);
+    this.app.use(bodyParser.urlencoded({ extended: true }));
+    this.app.use(bodyParser.json());
+    this.app.use(morgan('dev', { skip: () => !Logger.shouldLog }));
+    RegisterRoutes(this.app);
+    this.app.use(ErrorHandler.handleError);
+
+    // Disable linter as require statement is necessary for swagger files
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const swaggerDocument = require('../build/swagger.json');
+
+    this.app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
+  }
+
+  public async listen(port: number = this.port) {
+    process.on('uncaughtException', this.criticalErrorHandler);
+    process.on('unhandledRejection', this.criticalErrorHandler);
+    const listen = this.app.listen(this.port);
+    Logger.info(`${constants.environment} server running on port: ${this.port}`);
+    return listen;
+  }
+
+  private criticalErrorHandler(...args: unknown[]) {
+    Logger.error('Critical Error...', ...args);
+    process.exit(1);
+  }
+
+  private allowCors(req: express.Request, res: express.Response, next: express.NextFunction): void {
+    // In a later PR only specific sites will have access. For now, as we get setup, any site can have API access
```

Comment:
I find it helpful to mark future changes with a TODO tag, that way the codebase is easily searchable for "TODO" locations

Reply:
To be even more specific, it would be great if you made a github issue (e..g. "Restrict API site access") and did:
TODO(<link to issue>): issue description